packages/SwingSet/docs/garbage-collection.md

```diff
@@ -519,3 +474,121 @@ The delivery of each GC action is processed as follows:
   * deliver the message
     * the vat reacts to `dispatch.retireImport` by notifying any weak collections about the vref, which can delete the virtual entry indexed by it
     * this may provoke more drops or retirements
+
+# Comms Tracking
+
+Just as vats within a single SwingSet machine talk to each other through that machine's shared kernel, SwingSet machines talk to each other through the Internet.
+
+On each machine, the "comms vat" is responsible for managing all messages which leave that machine and travel to another. A single comms vat tracks multiple "remotes" (one per distant machine). From the comms point of view, the local kernel is one source of messages, which must be routed out to some number of remotes. A message which arrives from remote 1 might need to be forwarded to remote 2, or to the kernel, or to both (in the case of promise resolution). Comms has an object/promise table, with refcounts, very much like the kernel does. Comms has a c-list for each remote, plus an extra one facing the kernel, and translates object/promise identifiers between these remote- or kernel-facing- numberspaces and the comms internal ("local") numberspace.
+
+The comms vat does not use real JavaScript objects: there are no Remotables, Presences, Representatives, or Promises, which makes the code somewhat simpler than liveslots. However it must deal with asynchronous connections to the remotes, which makes it more complex than the kernel. When it sends a `dropExport` over the wire, that message won't be delivered immediately, and it might receive an arbitrary number of messages that were sent before the `dropExport` arrives. This complicates the reachability tracking.
+
+## Comms Tables
+
+The comms vat has an object table (key-value entries prefixed with `lo$NN`) that tracks an `.owner` for each object: the remote which exported that object (or empty to mean it came from the kernel). It also maintains `.reachable` and a `.recognizable` refcount for each. As with the kernel, these counts are only maintained for the downstream importers, not for the exporter. (Promises also have a refcount, but this only tracks usage by resolved promise data, not remotes: promises are retired when they are resolved, not when importers drop them).
+
+The comms vat has a c-list for each remote, plus one for the kernel. This c-list table has a special "isReachable" flag for each object. On the importer, this describes the state of the importer: `true` means they can reach (and recognize) the object, `false` means they can merely recognize it. On the exporter, this tracks whether we've send a `drop` or not.
```

Comment:
"On the exporter, this tracks whether we've send a `drop` or not" -- this is confusing.  Presumably `true` means reachable means have not sent a `drop`, which is inverted from what a reader might infer from reading that.

Reply:
changed to "On the exporter, it tracks whether we've send a `drop` or not: the flag is set to `true` until we send a `drop`, at which point it is cleared to `false`."